shared/common-adapters/wave-button.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react'
+import {Box2} from './box'
+import Icon from './icon'
+import Text from './text'
+import Button from './button'
+import Emoji from './emoji'
+import * as Styles from '../styles'
+import * as Container from '../util/container'
+import HiddenString from '../util/hidden-string'
+import * as Chat2Gen from '../actions/chat2-gen'
+import * as ChatTypes from '../constants/types/chat2'
+
+const Kb = {
+  Box2,
+  Icon,
+  Text,
+  Button,
+  Emoji,
+}
+
+type Recipient = { username: string } | { conversationIDKey: ChatTypes.ConversationIDKey }
```

Comment:
why not just use username always
if using a thing like this just add a `type` field instead of doing 'username' in recipient style checks

Reply:
> why not just use username always
want to send to a team in the next PR

Reply:
> why not just use username always

want to send to a team in a later PR

> if using a thing like this just add a type field instead of doing 'username' in recipient style checks

that's clunkier on the callsite, why is it better?

(something happened to previous attempt to make this GH comment)